sumo-function-utils/lib/datatransformer.js

```diff
@@ -10,55 +10,91 @@
 function Transformer() {
 }
 
-Transformer.prototype.azureAudit = function (data) {
+Transformer.prototype.azureAudit = function (context, data) {
 
     var finalResult = [];
     if (data instanceof Array) {
-        data.forEach( message => {
+        data.forEach(message => {
             if (message.records instanceof Array) {
                 message.records.forEach(subMsg => {
-                    finalResult.push(subMsg);
+                    finalResult.push(this.generateFormattedLog(context, subMsg));
                 });
             } else {
-                finalResult.push(message);
+                finalResult.push(this.generateFormattedLog(context, message));
             }
         });
     } else {
         if (data.records instanceof Array) {
-            data.records.forEach(subMsg => { finalResult.push(subMsg);});
-        } else { finalResult.push(data);}
+            data.records.forEach(subMsg => { finalResult.push(this.generateFormattedLog(context, subMsg)); });
+        } else { finalResult.push(this.generateFormattedLog(context, data)); }
     }
     return finalResult;
 };
 
+/**
+ * Function to generate a well formatted log to send to Sumo, following GCM's standards
+ * @param msg the original Azure log message
+ * @returns {object} of log following GCM's standard format
+ */
+Transformer.prototype.generateFormattedLog = function (context, msg) {
+    var log = {
+        level: this.getValue(() => msg.level),
+        host: this.getValue(() => msg.workflowId),
+        application: this.getValue(() => msg.properties.resource.workflowName),
+        environment: this.getValue(() => msg.properties.resource.resourceGroupName.split('-').slice(-1)[0]),
+        status: this.getValue(() => msg.properties.status),
+        process: this.getValue(() => msg.operationName),
+        trigger: this.getValue(() => msg.properties.resource.triggerName),
+        action: this.getValue(() => msg.properties.resource.actionName),
+        message: this.getValue(() => msg.properties.resource.actionName.replace(/_/g, ' ')),
+        runId: this.getValue(() => msg.properties.resource.runId)
+    };
```

Comment:
are there other props we should care about?

Reply:
lifecycle and lifecycleSuffix are there already :)
I used this file you provided as an example yesterday, and I just don't have the vpc property because I don't know what to put in it. Can I get it from the resourceGroupName value?